ignite/pkg/jsonfile/jsonfile.go

```diff
@@ -0,0 +1,311 @@
+	file := f.cache
+	if file != nil {
+		return file, nil
+	}
+	if err := f.Reset(); err != nil {
+		return nil, err
+	}
+	scanner := bufio.NewScanner(f.file)
+	for scanner.Scan() {
+		file = append(file, scanner.Bytes()...)
+	}
+	if err := scanner.Err(); err != nil {
+		return nil, err
+	}
+	f.cache = file
+	return file, nil
+}
+
+// Field return the param by key and the position into byte slice from the file reader.
+// Key can be a path to a nested parameter eg: app_state.staking.accounts
+func (f *JSONFile) Field(key string, param interface{}) error {
+	file, err := f.Bytes()
+	if err != nil {
+		return err
+	}
+
+	value, dataType, _, err := jsonparser.Get(file, strings.Split(key, keySeparator)...)
+	if err == jsonparser.KeyPathNotFoundError {
+		return ErrFieldNotFound
+	} else if err != nil {
+		return err
+	}
+
+	switch dataType {
+	case jsonparser.Boolean, jsonparser.Array, jsonparser.Number, jsonparser.Object:
+		err := json.Unmarshal(value, param)
+		if _, ok := err.(*json.UnmarshalTypeError); ok {
+			return ErrInvalidValueType
+		} else if err != nil {
+			return err
+		}
+	case jsonparser.String:
+		paramStr, ok := param.(*string)
+		if !ok {
+			return ErrInvalidValueType
+		}
+		*paramStr, err = jsonparser.ParseString(value)
+		if err != nil {
+			return err
+		}
+	default:
+		return ErrInvalidValueType
+	}
+	return nil
+}
+
+// WithKeyValue update a file value object by key
+func WithKeyValue(key string, value string) UpdateFileOption {
+	return func(update map[string][]byte) {
+		update[key] = []byte(`"` + value + `"`)
+	}
+}
+
+// WithTime update a time value
+func WithTime(key string, t int64) UpdateFileOption {
+	return func(update map[string][]byte) {
+		formatted := time.Unix(t, 0).UTC().Format(time.RFC3339Nano)
+		update[key] = []byte(`"` + formatted + `"`)
+	}
+}
+
+// WithKeyIntValue update a file int value object by key
+func WithKeyIntValue(key string, value int) UpdateFileOption {
+	return func(update map[string][]byte) {
+		update[key] = []byte(strconv.Itoa(value))
+	}
+}
+
+// Update updates the file with the new parameters by key
+func (f *JSONFile) Update(opts ...UpdateFileOption) error {
+	for _, opt := range opts {
+		opt(f.updates)
+	}
+	if err := f.Reset(); err != nil {
+		return err
+	}
+	_, err := io.Copy(f, f.file)
+	return err
+}
+
+// Write implement the write method for io.Writer interface
+func (f *JSONFile) Write(p []byte) (int, error) {
+	var err error
+	length := len(p)
+	for key, value := range f.updates {
+		p, err = jsonparser.Set(p, value, strings.Split(key, keySeparator)...)
+		if err != nil {
+			return 0, err
+		}
+		delete(f.updates, key)
+	}
+	f.cache = p
+
+	err = truncate(f.file, 0)
+	if err != nil {
+		return 0, err
+	}
+
+	if err := f.Reset(); err != nil {
+		return 0, err
+	}
+	n, err := f.file.Write(p)
+	if err != nil {
+		return n, err
+	}
+
+	if n != len(p) {
+		return n, io.ErrShortWrite
+	}
+
+	// FIXME passing the new byte slice length throws an error
```

Comment:
this looks like a serious problem

Reply:
I forgot to remove it. The truncate solves the problem